File: 27.remove-element.cpp
/*
 * @lc app=leetcode id=27 lang=cpp
 *
 * [27] Remove Element
 */

// @lc code=start
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int pointer, i;
        pointer = i = 0;
        for (i; i < nums.size(); ++i) {
            if (nums[i] != val) {
                nums[pointer++] = nums[i];
            }
        }
        return pointer;
    }
};
// @lc code=end

File: 27-RemoveElement.java
/*
 * @lc app=leetcode id=27 lang=java
 *
 * [27] Remove Element
 */

// @lc code=start
class Solution {
  public int removeElement(int[] nums, int val) {
    int slow, fast, length;
    slow = fast = length = 0;
    for (; fast < nums.length; fast++) {
      if (nums[fast] != val) {
        nums[slow] = nums[fast];
        length++;
        slow++;
      }
    }
    return length;
  }
}
// @lc code=end

